src/sys/signal.rs

```diff
@@ -6,44 +6,125 @@ use {Errno, Result};
+    SIGTERM = libc::SIGTERM as isize,
+    #[cfg(not(target_os = "macos"))]
+    SIGSTKFLT = libc::SIGSTKFLT as isize,
+    SIGCHLD = libc::SIGCHLD as isize,
+    SIGCONT = libc::SIGCONT as isize,
+    SIGSTOP = libc::SIGSTOP as isize,
+    SIGTSTP = libc::SIGTSTP as isize,
+    SIGTTIN = libc::SIGTTIN as isize,
+    SIGTTOU = libc::SIGTTOU as isize,
+    SIGURG = libc::SIGURG as isize,
+    SIGXCPU = libc::SIGXCPU as isize,
+    SIGXFSZ = libc::SIGXFSZ as isize,
+    SIGVTALRM = libc::SIGVTALRM as isize,
+    SIGPROF = libc::SIGPROF as isize,
+    SIGWINCH = libc::SIGWINCH as isize,
+    SIGIO = libc::SIGIO as isize,
+    #[cfg(not(target_os = "macos"))]
+    SIGPWR = libc::SIGPWR as isize,
+    SIGSYS = libc::SIGSYS as isize,
+    #[cfg(target_os = "macos")]
+    SIGEMT = libc::SIGEMT as isize,
+    #[cfg(target_os = "macos")]
+    SIGINFO = libc::SIGEMT as isize,
+}
+
+// TODO: NOT PUBLIC :/
+pub type SigNum = libc::c_int;
+
+// TODO: NOT PUBLIC :/
+pub fn signal_from(signum: SigNum) -> Signal {
+    match signum {
+        libc::SIGHUP => SIGHUP,
+        libc::SIGINT => SIGINT,
+        libc::SIGQUIT => SIGQUIT,
+        libc::SIGILL => SIGILL,
+        libc::SIGTRAP => SIGTRAP,
+        libc::SIGABRT => SIGABRT,
+        libc::SIGBUS => SIGBUS,
+        libc::SIGFPE => SIGFPE,
+        libc::SIGKILL => SIGKILL,
+        libc::SIGUSR1 => SIGUSR1,
+        libc::SIGSEGV => SIGSEGV,
+        libc::SIGUSR2 => SIGUSR2,
+        libc::SIGPIPE => SIGPIPE,
+        libc::SIGALRM => SIGALRM,
+        libc::SIGTERM => SIGTERM,
+        libc::SIGCHLD => SIGCHLD,
+        libc::SIGCONT => SIGCONT,
+        libc::SIGSTOP => SIGSTOP,
+        libc::SIGTSTP => SIGTSTP,
+        libc::SIGTTIN => SIGTTIN,
+        libc::SIGTTOU => SIGTTOU,
+        libc::SIGURG => SIGURG,
+        libc::SIGXCPU => SIGXCPU,
+        libc::SIGXFSZ => SIGXFSZ,
+        libc::SIGVTALRM => SIGVTALRM,
+        libc::SIGPROF => SIGPROF,
+        libc::SIGWINCH => SIGWINCH,
+        libc::SIGIO => SIGIO,
+        libc::SIGSYS => SIGSYS,
+        #[cfg(not(target_os = "macos"))]
+        libc::SIGPWR => SIGPWR,
+        #[cfg(not(target_os = "macos"))]
+        libc::SIGSTKFLT => SIGSTKFLT,
+        #[cfg(target_os = "macos")]
+        libc::SIGEMT => SIGEMT,
+        #[cfg(target_os = "macos")]
+        libc::SIGINFO => SIGINFO,
+        _ => panic!("There is no signal represented by {}.", signum),
+    }
+}
+
+
+pub use self::Signal::*;
+
+pub const SIGIOT : Signal = SIGABRT;
+pub const SIGPOLL : Signal = SIGIO;
+pub const SIGUNUSED : Signal = SIGSYS;
 
 pub const NSIG: libc::c_int = 32;
 
+struct SignalIterator {
+    signum : libc::c_int,
+}
+
+impl Iterator for SignalIterator {
+    type Item = Signal;
+    fn next(&mut self) -> Option<Signal> {
+        if self.signum < NSIG - 1 {
+            self.signum += 1;
+            Some(signal_from(self.signum))
+        } else {
+            None
+        }
+    }
+}
+
+impl Signal {
+    fn iterator() -> SignalIterator {
```

Comment:
I thought about the name as well, though I think iterator is pretty descriptive.